src/datachain/lib/file.py

```diff
@@ -486,13 +504,223 @@ class ImageFile(File):
     def read(self):
         """Returns `PIL.Image.Image` object."""
         fobj = super().read()
-        return Image.open(BytesIO(fobj))
+        return PilImage.open(BytesIO(fobj))
 
     def save(self, destination: str):
         """Writes it's content to destination"""
         self.read().save(destination)
 
 
+class Image(DataModel):
+    """`DataModel` for image file meta information."""
+
+    width: int = Field(default=-1)
+    height: int = Field(default=-1)
+    format: str = Field(default="")
+
+
+class VideoFile(File):
+    """`DataModel` for reading video files."""
+
+    def get_info(self) -> "Video":
+        """Returns video file information."""
+        from .video import video_info
+
+        return video_info(self)
+
+    def get_frame_np(self, frame: int) -> "ndarray":
+        """
+        Reads video frame from a file.
+
+        Args:
+            frame (int): Frame number to read.
+
+        Returns:
+            ndarray: Video frame.
+        """
+        from .video import video_frame_np
+
+        return video_frame_np(self, frame)
+
+    def get_frame(self, frame: int, format: str = "jpg") -> bytes:
+        """
+        Reads video frame from a file and returns as image bytes.
+
+        Args:
+            frame (int): Frame number to read.
+            format (str): Image format (default: 'jpg').
+
+        Returns:
+            bytes: Video frame image as bytes.
+        """
+        from .video import video_frame
+
+        return video_frame(self, frame, format)
+
+    def save_frame(self, frame: int, output_file: str) -> "VideoFrame":
+        """
+        Saves video frame as an image file.
+
+        Args:
+            frame (int): Frame number to read.
+            output_file (str): Output file path.
```

Comment:
what format is default? does it support different formats?

Reply:
Output format is taken from output file extension. See [here](https://github.com/iterative/datachain/pull/814/files#diff-281c712b90d682b4f700b6dcb351cd72e823440f3f19bae1d93d652e86279719R143-R146).

Reply:
so, extension determines it? I wonder if we need to clarify or will be kinda expected by end users 🤔

Reply:
Updated:
```
    def save_frame(
        self,
        frame: int,
        output_file: str,
        format: Optional[str] = None,
    ) -> "VideoFrame":
        """
        Saves video frame as an image file.

        Args:
            frame (int): Frame number to read.
            output_file (str): Output file path.
            format (str): Image format (default: use output file extension).

        Returns:
            VideoFrame: Video frame model.
        """
```